frontend-react/src/components/shared/Table/Table.tsx

```diff
@@ -0,0 +1,225 @@
+import classnames from "classnames";
+import sanitizeHtml from "sanitize-html";
+import { useMemo, useState } from "react";
+import { Icon } from "@trussworks/react-uswds";
+
+import { convertStringToSortableRepresentation } from "../../../utils/misc";
+
+import styles from "./Table.module.scss";
+
+const FilterOptionsEnum = {
+    NONE: "none",
+    ASC: "asc",
+    DESC: "desc",
+} as const;
+
+interface SortableTableHeaderProps {
+    columnTitle: string;
+    activeColumn: string;
+    sortOrder: string;
+    setSortOrder: (order: string) => void;
+    setActiveColumn: (column: string) => void;
+}
+
+interface RowData {
+    columnKey: string;
+    content: string;
+    modalContent?: string;
+}
+
+export interface TableProps {
+    borderless?: boolean;
+    compact?: boolean;
+    emptyState?: boolean;
+    fullWidth?: boolean;
+    scrollable?: boolean;
+    sortable?: boolean;
+    stackedStyle?: "default" | "headers";
+    striped?: boolean;
+    rowData: RowData[][];
+}
+
+const SortableTableHeader = ({
+    columnTitle,
+    activeColumn,
+    sortOrder,
+    setSortOrder,
```

Comment:
Let's use the naming pattern of `on<entity><action>` for callback props -- something like `onSortOrderChange`, `onActiveColumnChange`?

Reply:
Sounds good! Making that change.

Reply:
Hmmmm....can you tell me the full `useState` you expect? The first value is just a static state variable so `onSortOrderChange` sounds more like the callback itself, but the second parameter for `useState` must start with `set`, so `const [onSortOrderChange, setOnSortOrderChange] = useState("")` looks a little strange to me? As if both parameters are callback functions.

Reply:
Updated based on our conversation today!